### PR TITLE
Convert assessment_instances_select_for_auto_finish sproc to query

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Tested locally by reducing the frequency of the cron job and the age when auto-close happens.

### CODE REVIEW

File: apps/prairielearn/src/cron/autoFinishExams.ts [hunk 1]
```diff
@@ -16,29 +19,33 @@ import { config } from '../lib/config';
  * @see assessment.gradeAssessmentInstance
  */
 export async function run() {
-  const params = [config.autoFinishAgeMins];
-  const result = await sqldb.callAsync('assessment_instances_select_for_auto_finish', params);
-  const examList = result.rows;
+  const examList = await sqldb.queryRows(
```
Comment (nwalters512) [inline apps/prairielearn/src/cron/autoFinishExams.ts:22]: Nit: as long as we're refactoring, call this `assessments`?

File: apps/prairielearn/src/cron/autoFinishExams.ts [hunk 2]
```diff
@@ -16,29 +19,33 @@ import { config } from '../lib/config';
...
+    sql.select_assessments_to_auto_close,
+    { age_minutes: config.autoFinishAgeMins },
+    AssessmentInstanceSchema.pick({ id: true, open: true, assessment_id: true }),
+  );
 
   await async.eachSeries(examList, async (examItem) => {
```
Comment (nwalters512) [inline apps/prairielearn/src/cron/autoFinishExams.ts:28]: Nit, we could use a normal `for ... of ...` loop now that we're not using callbacks.

**Review by nwalters512 — APPROVED** [review]

File: apps/prairielearn/src/cron/autoFinishExams.ts (hunk 1 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/cron/autoFinishExams.ts]: Renamed to assessment_instances in f04ef40.

File: apps/prairielearn/src/cron/autoFinishExams.ts [hunk 3]
```diff
@@ -16,40 +18,44 @@ import { config } from '../lib/config';
...
         authn_user_id,
         requireOpen,
-        examItem.close_assessment,
+        close,
         overrideGradeRate,
         clientFingerprintId,
       );
     } catch (err) {
-      logger.error('Error finishing exam', error.addData(err, { examItem }));
+      logger.error('Error finishing exam', error.addData(err, { examItem: assessment_instance }));
```
Comment (nwalters512) [inline apps/prairielearn/src/cron/autoFinishExams.ts:52]: Nit:

```suggestion
      logger.error('Error finishing exam', error.addData(err, { assessment_instance }));
```

**Pull request merged**